x-pack/plugins/enterprise_search/server/routes/app_search/engines.test.ts

```diff
@@ -0,0 +1,177 @@
+      await handler(context, httpServerMock.createKibanaRequest(request), mockResponseFactory);
+    }
+
+    function executeRouteValidation(data: { query: object }) {
+      const [config] = router.get.mock.calls[0];
+      const validate = config.validate as RouteValidatorConfig<{}, {}, {}>;
+      const query = validate.query as ObjectType;
+      query.validate(data.query);
+    }
+
+    beforeEach(() => {
+      jest.resetAllMocks();
+      router = mockRouter.create();
+      registerEnginesRoute({
+        router,
+        config: {
+          host: APP_SEARCH_HOST_CONFIG,
+        },
+      });
+    });
+
+    describe('when the underlying App Search API returns a 200', () => {
+      const appSearchEnginesResponse = { name: 'engine1' };
+
+      beforeEach(() => {
+        AppSearchAPI.shouldBeCalledWith(
+          `${APP_SEARCH_HOST_CONFIG}/as/engines/collection?type=${TYPE}&page[current]=${PAGE_INDEX}&page[size]=10`,
+          {
+            headers: { Authorization: AUTH_HEADER },
+          }
+        ).andReturn(appSearchEnginesResponse);
+      });
+
+      it('should return 200 with a list of engines from the App Search API', async () => {
+        await callThisRoute();
+
+        expectResponseToBe200With({
+          body: appSearchEnginesResponse,
+          headers: { 'content-type': 'application/json' },
+        });
+      });
+    });
+
+    describe('when the underlying App Search API redirects to /login', () => {
+      beforeEach(() => {
+        AppSearchAPI.shouldBeCalledWith(
+          `${APP_SEARCH_HOST_CONFIG}/as/engines/collection?type=${TYPE}&page[current]=${PAGE_INDEX}&page[size]=10`,
+          {
+            headers: { Authorization: AUTH_HEADER },
+          }
+        ).andReturnRedirect();
+      });
+
+      it('should return 200 with a message', async () => {
+        await callThisRoute();
+
+        expectResponseToBe200With({
+          body: { message: 'no-as-account' },
+          headers: { 'content-type': 'application/json' },
+        });
+      });
+    });
+
+    describe('validation', () => {
+      function itShouldValidate(request: { query: object }) {
+        it('should be validate', async () => {
+          expect(() => executeRouteValidation(request)).not.toThrow();
+        });
+      }
+
+      function itShouldThrow(request: { query: object }) {
+        it('should throw', async () => {
+          expect(() => executeRouteValidation(request)).toThrow();
+        });
+      }
+
+      describe('when query is valid', () => {
+        const request = { query: { type: 'indexed', pageIndex: 1 } };
+        itShouldValidate(request);
+      });
+
+      describe('pageIndex is wrong type', () => {
+        const request = { query: { type: 'indexed', pageIndex: 'indexed' } };
+        itShouldThrow(request);
+      });
+
+      describe('type is wrong type', () => {
+        const request = { query: { type: 1, pageIndex: 1 } };
+        itShouldThrow(request);
+      });
```

Comment:
Looking at this test, I'm actually wondering now if we should change line 16 in engines.ts to an enum of either `indexed` or `meta` strings, and write a test for that accordingly (200, 200, and error for a type of 'foobar' for example). Thoughts? Do you think the enum is more helpful?

Reply:
My instinct is that it's unnecessary, since it's mainly just an internal endpoint that our front-end uses.

Reply:
Alrighty, no worries. I might come back to that later some day if I'm feeling super extra. 👍